### PR TITLE
X509: always base64-encode extensions for which _getMapping returns a bool

### PR DESCRIPTION
**Pull request by terrafrost**
Same thing as #628 but against 1.0 instead.

### CODE REVIEW

File: tests/Unit/File/X509/X509Test.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+oRYl21Q8EnvS71CemxEUkSsKJmfJKkQNCsOjc8AbX/V/X9R7LJkH3UEx6K2zQQKK
+k6m17mi63YW/+iPCGOWZ2qXmY5HPEyyF2L4L4IDryFJ+8xLyw3pH9/yp5aHZDtp6
+833K6qyjgHJT+fUzSEYpiwF5rSBJIGClOCY=
+-----END CERTIFICATE-----';
+
+        $x509 = new File_X509();
+
+        $spkac = $x509->loadX509($test);
+
+        $this->assertEquals($a['tbsCertificate']['extensions'][8]['extnValue'], 'MDUwDgYIKoZIhvcNAwICAgCAMA4GCCqGSIb3DQMEAgIAgDAHBgUrDgMCBzAKBggqhkiG9w0DBw==');
```
Comment (bantu) [inline tests/Unit/File/X509/X509Test.php]: Swap arguments. $expected comes first.

**Comment by terrafrost** [comment]
Replaces #628

**Comment by bantu** [comment]
> Btw, I am still curious if it is possible to change the branch a PR is made against. Any ideas?

Unfortunately it's not.

File: tests/Unit/File/X509/X509Test.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+oRYl21Q8EnvS71CemxEUkSsKJmfJKkQNCsOjc8AbX/V/X9R7LJkH3UEx6K2zQQKK
+k6m17mi63YW/+iPCGOWZ2qXmY5HPEyyF2L4L4IDryFJ+8xLyw3pH9/yp5aHZDtp6
+833K6qyjgHJT+fUzSEYpiwF5rSBJIGClOCY=
+-----END CERTIFICATE-----';
+
+        $x509 = new File_X509();
+
+        $cert = $x509->loadX509($test);
+
+        $this->assertEquals($cert['tbsCertificate']['extensions'][8]['extnValue'], 'MDUwDgYIKoZIhvcNAwICAgCAMA4GCCqGSIb3DQMEAgIAgDAHBgUrDgMCBzAKBggqhkiG9w0DBw==');
```
Comment (bantu) [inline tests/Unit/File/X509/X509Test.php]: Please swap the parameters. $expected comes first.

**Comment by terrafrost** [comment]
The latest commit addresses your comment.

I'll do the unit test for #625 after this has been merged (rather than creating a second copy of X509Test.php that's pretty much guaranteed to have a merge conflict)

**Comment by bantu** [comment]
This looks good. The test fails without the patch. :-)

**Comment by bantu** [comment]
Changed class name on merge.

**Pull request merged**